src/pages/[...slug].tsx

```diff
@@ -107,10 +108,17 @@ export const getStaticProps: GetStaticProps<Props, Params> = async (
   const mdPath = join("/content", ...params.slug)
   const mdDir = join("public", mdPath)
 
+  let tocNodeItems: TocNodeType[] = []
+  const tocCallback = (toc: TocNodeType): void => {
```

Comment:
should be defined on `toc.ts`

Reply:
This function needs `tocNodeItems` in scope, and we need this inside this function to pass it as props to the layout/page. We have to use a side-effect here, since we have no other way to capture a return value from this rehype plugin... I think we need to keep this here.